pkg/controller/pkg/manager/pod.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manager
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	runAsUser                = int64(2000)
+	runAsGroup               = int64(2000)
+	allowPrivilegeEscalation = false
+	privileged               = false
+	runAsNonRoot             = true
+)
+
+// buildInstallPod builds a Pod that parses and outputs a package.
+func buildInstallPod(name, namespace, img string, labels map[string]string, imagePullSecrets []corev1.LocalObjectReference, imagePullPolicy *corev1.PullPolicy) *corev1.Pod {
+	pullPolicy := corev1.PullIfNotPresent
+	if imagePullPolicy != nil {
+		pullPolicy = *imagePullPolicy
+	}
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    labels,
+		},
+		Spec: corev1.PodSpec{
+			SecurityContext: &corev1.PodSecurityContext{
+				RunAsNonRoot: &runAsNonRoot,
+				RunAsUser:    &runAsUser,
+				RunAsGroup:   &runAsGroup,
+			},
+			ImagePullSecrets: imagePullSecrets,
+			RestartPolicy:    corev1.RestartPolicyNever,
+			Containers: []corev1.Container{
+				{
+					Name:            "unpack",
+					Image:           img,
+					ImagePullPolicy: pullPolicy,
+					SecurityContext: &corev1.SecurityContext{
+						RunAsUser:                &runAsUser,
+						RunAsGroup:               &runAsGroup,
+						AllowPrivilegeEscalation: &allowPrivilegeEscalation,
+						Privileged:               &privileged,
+						RunAsNonRoot:             &runAsNonRoot,
```

Comment:
We do not need to. A container will inherit from Pod if it is not set explicitly on the container. However, not all fields are available on the Pod, such as `Privileged` and `AllowPrivilegeEscalation`. If we wanted to strip some out here we should remove them from the Pod security context. I actually kind of like having them explicitly defined at both levels though because if someone is adjusting them in the future it feels like a stronger check if they have to change a value rather than add something net new (once again, we would have them defined at the container level and adding something to Pod security context would not override, so not super risky). However, this is mostly just a philosophical thing, so happy to simplify if desired.